packages/demo/src/components/PreviewDimension.tsx

```diff
@@ -0,0 +1,77 @@
+import { Button, Input, Stack, Typography } from '@mui/material'
+import { PluginComponent } from './FieldPluginDemo'
+import { useState } from 'react'
+
+export const PreviewDimension: PluginComponent = (props) => {
+  const { data, actions } = props
+
+  const label =
+    typeof data.content === 'undefined'
+      ? 'undefined'
+      : JSON.stringify(data.content)
+
+  const [width, setWidth] = useState(300)
+
+  const handleClickMobile = () => {
+    actions.setPreviewDimension({
+      tag: 'mobile',
+    })
+  }
+  const handleClickTablet = () => {
+    actions.setPreviewDimension({
+      tag: 'tablet',
+    })
+  }
+  const handleClickDesktop = () => {
+    actions.setPreviewDimension({
+      tag: 'desktop',
+    })
+  }
+  const handleClickCustom = () => {
+    actions.setPreviewDimension({
+      tag: 'custom',
+      width: width,
+    })
+  }
+
+  return (
+    <Stack gap={2}>
+      <Typography variant="subtitle1">Field Value</Typography>
+      <Typography textAlign="center">{label}</Typography>
+      <Button
+        variant="outlined"
+        color="secondary"
+        onClick={handleClickMobile}
+      >
+        Mobile
+      </Button>
+      <Button
+        variant="outlined"
+        color="secondary"
+        onClick={handleClickTablet}
+      >
+        Tablet
+      </Button>
+      <Button
+        variant="outlined"
+        color="secondary"
+        onClick={handleClickDesktop}
+      >
+        Desktop
+      </Button>
+      <Button
+        variant="outlined"
+        color="secondary"
+        onClick={handleClickCustom}
+      >
+        Custom
+      </Button>
+      <Input
+        type="number"
+        value={width}
+        onChange={(e) => setWidth(Number(e.target.value))}
+        placeholder="Custom Width"
+      />
```

Comment:
Since the `width` input and the `custom` button are related I will group them visually too, so it is clear that they are related:

Now it looks like:
<img width="731" height="145" alt="Screenshot 2025-07-23 at 13 07 00" src="https://github.com/user-attachments/assets/cb54a8ea-a4be-442a-b0e8-dce6a0838252" />

I will suggest something like:
<img width="700" height="114" alt="Screenshot 2025-07-23 at 13 10 02" src="https://github.com/user-attachments/assets/3c7faf28-859d-4ce0-86a2-8c4ad370e400" />